src/cdk/table/table.ts

```diff
@@ -105,9 +107,25 @@ export interface RowOutlet {
  * Union of the types that can be set as the data source for a `CdkTable`.
  * @docs-private
  */
-type CdkTableDataSourceInput<T> =
+export type _CdkTableDataSourceInput<T> =
     DataSource<T>|Observable<ReadonlyArray<T>|T[]>|ReadonlyArray<T>|T[];
 
+/**
+ * Collection viewer for the `CdkTable`. For backwards compatibility, the collection viewer must
+ * be a `BehaviorSubject`.
```

Comment:
We can potentially change this; I assume that this isn't going to be widely used inside Google since `viewChange` has just been a placeholder

Reply:
Internal targets could be quickly migrated. What about external apps?

Reply:
For a behavior change like this, I think this is something we could change for a major version (which we're currently in the window for) so long as we document it.

Reply:
@jelbourn Following up since it's been a while. Is now a still a good time to make this change?

Reply:
What's the benefit of changing it?

Reply:
The behavior subject emits a range the spans the entire data set when the table is initialized. This causes the table to briefly render all rows, then truncate them once it attaches to the virtual scroll viewport. By changing this to a regular `Subject`, we'd have more control over when the range is emitted.

For now, I've hacked around it by overriding the behavior subject via provider. Then emitting an empty range as the initial value. This prevents rendering the rows before the virtual scroll viewport is attached.

Reply:
We'd be able to change it starting in August 

Revisiting this, could you expand the comment to touch on why this interface is necessary? (IIRC it's so that the table can inject the viewChange behavior from another directive instead of completely implementing it itself).

Actually typing that out, it _might_ make sense to rename this to something named like `ViewChangeXxx` instead of `CollectionViewer` since `CollectionViewer` could potentially be expanded to include more stuff, but that's not super important with the features it has today.